Rovio always fails to compile the first time on a new computer

This is something I have seen on a few systems now and am unsure what is causing it. If you create a new catkin_ws and build rovio it will always fail the first time with the error:

Errors     << rovio:make /home/zac/catkin_ws/logs/rovio/build.make.000.log     
In file included from /home/zac/catkin_ws/src/rovio/src/rovio_node.cpp:40:0:
/home/zac/catkin_ws/src/rovio/include/rovio/RovioNode.hpp:50:34: fatal error: rovio/SrvResetToPose.h: No such file or directory
 #include <rovio/SrvResetToPose.h>
                                  ^
compilation terminated.
make[2]: *** [CMakeFiles/rovio.dir/src/rovio_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/rovio.dir/all] Error 2
make: *** [all] Error 2
cd /home/zac/catkin_ws/build/rovio; catkin build --get-env rovio | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -

However, if the build is immediately rerun everything builds fine and the error never appears again.

Hi,
I get the same error, even though I'm using the latest version of master with the dependency on the exported tag:

Errors     << rovio:make /home/marco/catkin_ws/logs/rovio/build.make.000.log                                                                                                              
In file included from /home/marco/catkin_ws/src/rovio/src/rovio_node.cpp:40:0:
/home/marco/catkin_ws/src/rovio/include/rovio/RovioNode.hpp:50:34: fatal error: rovio/SrvResetToPose.h: No such file or directory
 #include <rovio/SrvResetToPose.h>
                                  ^
compilation terminated.
make[2]: *** [CMakeFiles/rovio.dir/src/rovio_node.cpp.o] Error 1
make[1]: *** [CMakeFiles/rovio.dir/all] Error 2
make: *** [all] Error 2
cd /home/marco/catkin_ws/build/rovio; catkin build --get-env rovio | catkin env -si  /usr/bin/make --jobserver-fds=6,7 -j; cd -

Again, as Zac mentioned, if I compile rovio again after the error everything's fine.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -107,16 +107,17 @@ else()
 endif()
 target_link_libraries(${PROJECT_NAME} ${catkin_LIBRARIES} ${YamlCpp_LIBRARIES} ${OpenMP_EXE_LINKER_FLAGS} ${OPENGL_LIBRARIES} ${GLUT_LIBRARY} ${GLEW_LIBRARY})
 
-add_dependencies(${PROJECT_NAME}  ${catkin_EXPORTED_TARGETS})
-
 add_executable(rovio_node src/rovio_node.cpp)
 target_link_libraries(rovio_node ${PROJECT_NAME})
+add_dependencies(rovio_node ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 
 add_executable(rovio_rosbag_loader src/rovio_rosbag_loader.cpp)
 target_link_libraries(rovio_rosbag_loader ${PROJECT_NAME})
+add_dependencies(rovio_rosbag_loader ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 
 add_executable(feature_tracker_node src/feature_tracker_node.cpp)
 target_link_libraries(feature_tracker_node ${PROJECT_NAME})
+add_dependencies(feature_tracker_node ${${PROJECT_NAME}_EXPORTED_TARGETS} ${catkin_EXPORTED_TARGETS})
 
 if(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/gtest/")
 	message(STATUS "Building GTests!")

